### DISCUSSION TITLE
Questions on Past alerts + geometry props

### DISCUSSION THREAD

**Discussion by indracha-maersk** [weather-gov/api#789] [General]
Hello there, 

Couple of questions,
- The responses we get from alerts/active endpoint, all seem to have '"geometry:":null'. Is this expected? Aren't we supposed to get some geometry value to identify the area?
- How do I get past alert? For instance, I am particularly interested in getting old alerts (last 1 month) related to Tsunami (warning/advisory).

Appreciate your help in responding to my questions.
Have a great day ahead!

**Comment by sullynole** [comment]
api.weather.gov/alerts has alerts from the past 7 days. 

https://www.weather.gov/documentation/services-web-api
The examples tab has more details about alerts; recommend reviewing this document about geolocation: https://www.weather.gov/media/documentation/docs/NWS_Geolocation.pdf

**Comment by dwhitemv25** [comment]
About the geometry, we just had a question asked about this, see #786.

For older content, you want to find the parent text product. The product code for tsunami warnings is TSU.  Check the available sources at this URL: https://api.weather.gov/products/types/TSU/locations

**Comment by indracha-maersk** [comment]
Another question - Is this Tsunami information same as what being displayed on this page - https://ntwc.ncep.noaa.gov/
Basically, I want to show the coordinate where Tsunami is originating (lat,lon).

Do we get this point (not geometry or region or zones) in REST endpoint?

Many thanks for your response!

**Reply by dwhitemv25** [reply]
TSU is one of the text products (the stuff linked below the map). [This page](https://ntwc.ncep.noaa.gov/?page=product_list) has all the message types that NTWC generates -- the product code is the first 3 letters under the 'NWS AWIPS ID' column, and the last 3 is the location.

Looks like there is a machine parseable location (lat/lon) in the CAP format alerts and in the RSS feed. Check under the 'Follow us' item in the sidebar on the NTWC site for the links.

I don't believe the NWS API (the one supported by this site) offer any endpoints that provide JSON format data for tsunamis.

**Comment by 6TELOIV** [comment]
I've found that alerts for localized events have a specific `geometry` value (for example, flood warnings along a certain area of a river), while other alerts for entire regions (for example, tornado watches, thunderstorms, etc) use `geocode` and `affectedZones`. `affectedZones` will be an array of URLs for the zones, and `GET`ing those will give you geojson data that you can use to display the affected area.
